mat64/format.go

```diff
@@ -9,15 +9,46 @@ import (
 	"strconv"
 )
 
-// Format prints a pretty representation of m to the fs io.Writer. The format character c
+// Formatted returns a fmt.Formatter for the matrix m based on the provided prefix, margin and
```

Comment:
This is a lot nicer. Is the prepend for '\t' or something? In the json case often json  has no newlines or tabs. For matrix, I assume we're adding braces/ | around the matrix?

In my thinking, "dot" does not replace zero elements, but unused elements. For example, symmetric may have no elements that are zero but has half of the elements unused.